packages/loader/container-loader/src/collabWindowTracker.ts

```diff
@@ -69,9 +69,16 @@ export class CollabWindowTracker {
         }
 
         this.opsCountSinceNoop++;
-        if (this.opsCountSinceNoop >= this.NoopCountFrequency) {
-            this.submitNoop(false /* immediate */);
-            return;
+        if (this.opsCountSinceNoop === this.NoopCountFrequency) {
```

Comment:
Can you please add a comment above it telling why we do promise dance (to ensure that we send one noop after a batch of many ops). Thanks!